subprojects/common/report-viewer/src/main/kotlin/com/avito/report/internal/ReportFileProviderImpl.kt

```diff
@@ -0,0 +1,49 @@
+package com.avito.report.internal
+
+import com.avito.android.Result
+import com.avito.report.ReportFileProvider
+import java.io.File
+import java.util.UUID
+
+class ReportFileProviderImpl(
+    override val rootDir: Lazy<File>,
+    private val className: String,
+    private val methodName: String,
+    private val randomFileNameGenerator: () -> String = { UUID.randomUUID().toString() }
+) : ReportFileProvider {
+
+    // todo should be passed with instrumentation params, see [ArtifactsTestListener]
+    private val runnerOutputFolder = "runner"
+
+    // todo should be passed with instrumentation params, see [ReportViewerTestReporter]
+    private val reportFileName = "report.json"
+
+    override fun provideReportDir(): File {
+        val testFolderName = "$className#$methodName" // todo add deviceName
+
+        val runnerDirectory = File(rootDir.value, runnerOutputFolder)
+
+        return File(runnerDirectory, testFolderName)
+    }
+
+    override fun provideReportFile(): File {
+        return File(provideReportDir(), reportFileName)
+    }
+
+    override fun getFile(relativePath: String): File {
+        return File(provideReportDir(), relativePath)
+    }
```

Comment:
Is report in the same directory as other files?

I'm thinking about stricter layout to exclude this mixing at all:

```
myTest#test/
                         | report.json
                         | extras/
                                       |  file-for-fs-1.png
                                       |  file-for-fs-2.txt
```

Reply:
E.g. clients can write in the same directory by `generateFile` and can override accidentally.
This is why I want to distinguish randomly generated files.

Reply:
myTest#test dir should be unique

Reply:
report.json could be overwritten also in this case

Reply:
for now i don't know how to add device name in test name

Reply:
I mean possible clashes inside myTest#test dir.
It seems to me that we need isolated directories (destinations) for `generateFile` and `generateUniqueFile`.
It's just a hypothetical problem. It can be done later.